### ISSUE TITLE
Cache not working

### ISSUE THREAD

**Issue by glitsj16** [netblue30/fdns#4]
Hi, I'm currently testing fdns as my main DNS rsolver. I made a systemd service file to start fdns at boot:

```
[Unit]
Description=Firejail DNS-over-HTTPS Proxy Server
Documentation=man:fdns(1)
After=network.target
Before=network-online.target nss-lookup.target
Wants=nss-lookup.target

[Service]
ExecStart=/usr/bin/fdns --proxy-addr-any --server=powerdns

[Install]
WantedBy=multi-user.target
```

Everything seems to work great, except `caching`. Using _fdns --monitor_ I never see any changes in the cached items count. Running with _--debug_ didn't throw any light on this behavior. For the record, the OS is Arch Linux. Manually built fdns from git and I'm assuming/expecting running dig (on the same domain) twice would count the second run as cached and show a query time close to 0 msec. Instead I only see a reduced query time, but nothing close to what I would expect:

```
$ dig google.com

; <<>> DiG 9.14.7 <<>> google.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 7680
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;google.com.			IN	A

;; ANSWER SECTION:
google.com.		90	IN	A	172.217.26.46

;; Query time: 194 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Sat Nov 16 17:10:47 CET 2019
;; MSG SIZE  rcvd: 55


$ dig google.com

; <<>> DiG 9.14.7 <<>> google.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 7680
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;google.com.			IN	A

;; ANSWER SECTION:
google.com.		88	IN	A	172.217.26.46

;; Query time: 69 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Sat Nov 16 17:10:47 CET 2019
;; MSG SIZE  rcvd: 55
```
 Is something extra needed for caching to work? Or is this not implemented yet?

**Comment by netblue30** [comment]
I sent you an invite. Can you merge in the service file in etc directory (https://github.com/netblue30/fdns/tree/master/etc). I'll install it as default on make install. Thanks.

Interesting, the numbers (requests, cached) don't seem to increment in the monitor. On Wireshark, the requests seem to be going out over HTTPS/TLS. I am comparing it with ping, where ping works fine and dig doesn't. Caching works fine in ping. Bug!

**Comment by glitsj16** [comment]
Looks like the issue was on my side after all. While trying to debug #10 I decided to reset my `/etc/resolv.conf` (only retaining 'nameserver 127.0.0.1'). After doing that `fdns --monitor` happily started to report cached items. Lesson learned: _don't use esoteric options in /etc/resolv.conf while using fdns_.

I'll leave this open for a while to give other people ample opportunity to comment, but I'm quite sure the bug was not in the fdns code.

**Comment by netblue30** [comment]
Let me try again, I think I saw something strange with dig.

**Comment by rusty-snake** [comment]
I use a script to fetch remote from repos for all my local git repos, if I run this script fdns gives me the following logs (ignoring gitlab queries).

`fdns --workers=1`
```
github.com, encrypted
github.com (ipv6), dropped
github.com, cached
github.com (ipv6), dropped
... only cached
```

`fdns --workers=5`
```
github.com, encrypted
github.com (ipv6), dropped
github.com, cached
github.com (ipv6), dropped
github.com, encrypted
github.com (ipv6), dropped
github.com, encrypted
github.com (ipv6), dropped
github.com, cached
github.com (ipv6), dropped
github.com, encrypted
github.com (ipv6), dropped
github.com, cached
github.com (ipv6), dropped
github.com, encrypted
github.com (ipv6), dropped
github.com, cached
github.com (ipv6), dropped
... only cached
```

**Comment by netblue30** [comment]
closed for now
